core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java

```diff
@@ -45,6 +47,14 @@ public UpdateTableRequest(List<UpdateRequirement> requirements, List<MetadataUpd
     this.updates = updates;
   }
 
+  public UpdateTableRequest(
```

Comment:
I commented in #7750, but I don't think this needs to be public. I'd also not add a `forTable` method and instead pass the identifier in when creating the builder.

Reply:
I updated the constructor to be non-public. I removed `forTable()` for now and I think we can handle the identifier inside the builder in the other PRs, since we don't actually need it here